modules/build/src/main/scala/scala/build/input/Inputs.scala

```diff
@@ -313,8 +314,17 @@ object Inputs {
             if isShebangScript(String(content)) then
               s"$arg scripts with no file extension should be run with 'scala-cli shebang'"
             else
-              s"$arg: unrecognized source type (expected .scala or .sc extension, or a directory)"
-          else s"$arg: not found"
+              s"""$arg: unrecognized source type (expected .scala or .sc extension, or a directory),
+                 |if it's meant to be a script add a '!#' pointing to 'scala-cli shebang' in the top line
```

Comment:
also, I wonder if we should print the same error on Windows... 🤔 
`shebang` headers don't normally work there, unless used with `cygwin`...